contracts/governance/FeeSharingProxy/FeeSharingLogic.sol

```diff
@@ -297,6 +302,63 @@ contract FeeSharingLogic is SafeMath96, IFeeSharingProxy, Ownable, FeeSharingPro
         emit UserFeeWithdrawn(msg.sender, _receiver, _loanPoolToken, amount);
     }
 
+    /**
+     * @dev withdraw all of the RBTC balance based on particular checkpoints
+     *
+     * RBTC balance consists of:
+     * - rbtc balance
+     * - wrbtc balance which will be unwrapped to rbtc
+     * - iwrbtc balance which will be unwrapped to rbtc
+     *
+     */
+    function withdrawRBTC(uint32 _maxCheckpoints, address _receiver) external {
+        uint256 wrbtcAmount;
+        uint256 rbtcAmount;
+        uint256 iWrbtcAmount;
+        uint256 endRBTC;
+        uint256 endWRBTC;
+        uint256 endIWRBTC;
+        uint256 iWRBTCloanAmountPaid;
+        address user = msg.sender;
+
+        IWrbtcERC20 wrbtcToken = protocol.wrbtcToken();
+
+        address loanPoolTokenWRBTC = _getAndValidateLoanPoolWRBTC(address(wrbtcToken));
+
+        if (_receiver == ZERO_ADDRESS) {
+            _receiver = msg.sender;
+        }
+
+        (rbtcAmount, wrbtcAmount, iWrbtcAmount, endRBTC, endWRBTC, endIWRBTC) = _getRBTCBalances(
+            user,
+            _maxCheckpoints
+        );
+
+        processedCheckpoints[user][RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT] = endRBTC;
+        processedCheckpoints[user][address(wrbtcToken)] = endWRBTC;
+        processedCheckpoints[user][loanPoolTokenWRBTC] = endIWRBTC;
```

Comment:
previous algorythm required amount > 0 to write a checkpoint. why is it changed?

Reply:
updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/458/commits/dadad69a121b6a940b5f894715c22af0678deef5)